ADAL/tests/ADAcquireTokenTests.m

```diff
@@ -1650,4 +1666,45 @@ - (void)testResilencyTokenDeletion
     XCTAssertEqual(allItems.count, 0);
 }
 
+- (void)testAllowSilentRequestParameters
+{
+    ADAuthenticationContext* context = [self getTestAuthenticationContext];
+    ADRequestParameters *params = [[ADRequestParameters alloc] initWithAuthority:context.authority
+                                                                        resource:TEST_RESOURCE
+                                                                        clientId:TEST_CLIENT_ID
+                                                                     redirectUri:TEST_REDIRECT_URL.absoluteString
+                                                                      identifier:[ADUserIdentifier identifierWithId:TEST_USER_ID]
+                                                                      tokenCache:context.tokenCacheStore
+                                                                extendedLifetime:NO
+                                                                   correlationId:nil
+                                                              telemetryRequestId:nil];
+    ADAuthenticationRequest* req = [ADAuthenticationRequest requestWithContext:context requestParams:params error:nil];
+    [req setSilent:YES];
+    [req setAllowSilentRequests:YES];
+    
+    
+    // Following we add a mock response and specify the request url we expect (it must include login_hint)
+    ADTestURLResponse* response = [ADTestURLResponse requestURLString:@"https://login.windows.net/contoso.com/oauth2/authorize?client_id=c3c7f5e5-7153-44d4-90e6-329686d48d76&resource=resource&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&nux=1&response_type=code&login_hint=eric_cartman%40contoso.com&x-client-Ver=" ADAL_VERSION_STRING
+                                                    responseURLString:@"https://idontmatter.com"
+                                                         responseCode:401
+                                                     httpHeaderFields:@{}
+                                                     dictionaryAsJSON:@{}];
+    [ADTestURLConnection addResponse:response];
+    
+    // We send the actual silent network request
+    [req requestToken:^(ADAuthenticationResult *result)
+     {
+         // if request url is not expected,
+         // our network mock will fail the unit test and it won't hit here
+         XCTAssertNotNil(result);
```

Comment:
The thought here is because objc doesn't complain when you send a message to nil, it's possible for an assert to succeed even when the test should have failed. e.g.

```
XCTAssertEqual(result.status, 0)
```

Result could be `nil` and that assert would still pass because sending a message to nil also returns zero.  Checking for nils before checking result values can prevent tests passing that should have failed and make the reason why a test failed clearer.